### PR TITLE
chore!: deprecate ensure_installed=maintained

### PR DESCRIPTION
**Pull request by clason**
With the increasing number of parsers added to `nvim-treesitter`, the label "maintained" has stopped being a useful distinction, as just having a maintainer listed at time of inclusion does not entail any stability guarantee implied by this label. Users should explicitly specify the list of parsers that they want to use according to their risk tolerance, or use `ensure_installed = 'all'` if they insist on drinking from the firehose (not recommended!)

Deprecation period is until April 30; hopefully that will leave enough time for the ecosystem to adapt.
The warning is emitted from `parsers.maintained_parsers()` in the hope that this will be most visible (and not break CI).

@theHamsta @vigoux @kyazdani42

### CODE REVIEW

**Review by kyazdani42 — APPROVED** [review]

**Comment by clason** [comment]
(updated the documentation and README as well)

**Comment by clason** [comment]
OK, `ubuntu-latest` doesn't seem to have `vim.notify_once` o.O?

File: README.md
Comment (theHamsta) [inline README.md:155]: here is still `maintained`

File: README.md
Comment (theHamsta) [inline README.md:106]: I would not document deprecated stuff.

File: README.md
Comment (clason) [reply README.md:155]: Yes, on purpose, as the option is still there and works.

When the deprecation hits the fan, I will remove all mention of "maintained" from the repo with a hot `grep`.

File: README.md
Comment (clason) [reply README.md:106]: The idea was to sprinkle the deprecation warning liberally throughout the docs to increase the chances of people seeing it...

**Comment by clason** [comment]
I've removed all mention of `"maintained"` from the documentation.

**Comment by clason** [comment]
@theHamsta I'll merge (and announce) this after your approval.

**Review by theHamsta — APPROVED** [review]

**Pull request merged**
